src/radical/pilot/agent/radical-pilot-agent-multicore.py

```diff
@@ -2071,7 +2071,14 @@ def construct_command(self, task_exec, task_args, task_numcores,
             task_command = task_exec
 
         # Construct the hosts_string, env vars
-        hosts_string = ",".join([slot.split(':')[0] for slot in task_slots])
+        # On some Crays, like on ARCHER, the hostname is "archer_N".
+        # In that case we strip off the part upto and including the underscore.
+        #
+        # TODO: If this ever becomes a problem, i.e. we encounter "real" hostnames
+        #       with underscores in it, or other hostname mangling, we need to turn
+        #       this into a system specific regexp or so.
+        #
```

Comment:
We could eventually also make it a callout to 'host' ir 'nslookup', as the short names should always be resolvable on system level.  But right now I think this is the appropriate (if ugly) approach.